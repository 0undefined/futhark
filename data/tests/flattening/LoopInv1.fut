fun [int] addRows ([int] xs, [int] ys) =
  map(+, zip (xs,ys))

fun [[int]] main ([[int]] xss, [int] ys) =
  map(fn [int] ([int] xs) => addRows(xs,ys), xss)
