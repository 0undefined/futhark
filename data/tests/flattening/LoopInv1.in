[ [1,2,3], [4,5,6]
, [6,7,8], [9,10,11]
, [3,2,1], [4,5,6]
, [8,7,6], [11,10,9]
]
[1,2,3]
