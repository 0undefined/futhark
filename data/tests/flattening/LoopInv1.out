[[2, 4, 6],
 [5, 7, 9],
 [7, 9, 11],
 [10, 12, 14],
 [4, 4, 4],
 [5, 7, 9],
 [9, 9, 9],
 [12, 12, 12]]
